unix sockets, bugs/typos fixes, go.mod

VadimIpatov opened the pull request.

saschat commented on mock_conn.go:
```diff
@@ -20,7 +20,7 @@ type mockConn struct {
 // second request succeeds. The address has the following structure:
 // <id>-<failure-pattern>:<port> (port gets automatically inserted).
-func newMockConn(address, username, password string, config *Config) mcConn {
+func newMockConn(address, schema, username, password string, config *Config) mcConn {
```
schema -> scheme

VadimIpatov replied:
> schema -> scheme

fixed

saschat commented on server_conn.go:
```diff
@@ -70,24 +72,31 @@ func (sc *serverConn) performStats(m *msg) (McStats, error) {
 
 func (sc *serverConn) connect() error {
-	c, err := net.DialTimeout("tcp", sc.address, sc.config.ConnectionTimeout)
+	fmt.Printf("address = %s, scheme = %s\n", sc.address, sc.scheme)
```
Lets remove this debug print.

VadimIpatov replied:
> Lets remove this debug print.

fixed

VadimIpatov commented:
> Generally, this looks good to me. Just to tiny comments.
> 
> Could you maybe also modify the Travis configuration to remove the old versions that fail (I am fine with not supporting them, @alevy you?) and add newer versions that are missing.

Sure, I can do it, but I have a question. Due to go.mod was added, the earliest supported version of Go is 1.11. It's okay for you?

VadimIpatov commented:
> Generally, this looks good to me. Just to tiny comments.
> 
> Could you maybe also modify the Travis configuration to remove the old versions that fail (I am fine with not supporting them, @alevy you?) and add newer versions that are missing.

Sure, I've fixed it. Also, I've updated a bit the install and usage sections in the readme to let users know how to use module-aware mode. Please, add new tag v2.0.2 or smth to make it work.

VadimIpatov commented:
> Looks good to me. We'll tag a v3 since this doesn't seem necessarily backwards compatible with older versions of Go

In this case it's necessary to replace github.com/memcachier/mc/v2 to github.com/memcachier/mc/v3 in go.mod and README

VadimIpatov commented:
Please, replace github.com/memcachier/mc/v2 to github.com/memcachier/mc/v3 in go.mod and README. It doesn't work currently

saschat commented:
@VadimIpatov I am new to this way of go module versioning so I am a bit confused. I made the updates you suggested but it still does not work for me. Does it work for you?

VadimIpatov commented:
> @VadimIpatov I am new to this way of go module versioning so I am a bit confused. I made the updates you suggested but it still does not work for me. Does it work for you?

I supposed that we will use v2, but we decided to tag this version as v3.0.0. It caused the problem. I think the right way to fix it is to change v2 to v3 in go.mod (and readme too). 
FYI: https://medium.com/@adiach3nko/package-management-with-go-modules-the-pragmatic-guide-c831b4eaaf31
I've made these changes in my fork. If it's possible, you can just merge it again.

saschat commented:
I did change it from v2 to v3 in the README and go.mod on master but it is not working for me. Is it working for you?

saschat commented:
And by not working I mean

```
$ go get github.com/memcachier/mc/v3
go: downloading github.com/memcachier/mc v1.0.0
go: downloading github.com/memcachier/mc v2.0.1+incompatible
go get github.com/memcachier/mc/v3: module github.com/memcachier/mc@upgrade found (v2.0.1+incompatible), but does not contain package github.com/memcachier/mc/v3
```

VadimIpatov commented:
> I did change it from v2 to v3 in the README and go.mod on master but it is not working for me. Is it working for you?

try to mark the last commit as v3.0.1

saschat commented:
I did that already as well ;-)

VadimIpatov commented:
> I did that already as well ;-)

now I see it :) it works well for me now. thanks!

saschat commented:
You are right, it does work. I wonder why I got this error earlier (I did clear the module cache).

Anyway, thanks for the help and for the shared article.

VadimIpatov commented:
> You are right, it does work. I wonder why I got this error earlier (I did clear the module cache).
> 
> Anyway, thanks for the help and for the shared article.

You're welcome! :)
